File: CBox/model/Action.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.ComponentModel;

namespace cbox.model
{
    public class Action
    {
        public int Ident;
        public BindingList<string> Yield = new BindingList<string>();
        public string Title { get; set; }
        public BindingList<string> TargetClasses = new BindingList<string>();
        public BindingList<string> RevealsClasses = new BindingList<string>();

        public int Time { get; set; }
        public double Risk { get; set; }
        public double Pain { get; set; }
        public int Money { get; set; }

        public bool VisibleHeadlineHint{ get; set; }

        public string TitleWithClassnames
        {
            get
            {
                return Title + " <" + string.Join(", ", TargetClasses) + ">";
            }
        }
    }
}

File: CBoxClient/lib/model/Action.ts



module cbox {

    export class Action {

        cost: number;
        ident: number;
        pain: number;
        risk: number;
        target_classes: string[] = [];
        time_used_occupied: number;
        time_used_process: number;
        title: string;

        public static fromObject(obj: {}) {
            var a = new Action();

            a.title = obj['Title'];
            a.cost = parseFloat(obj['Cost']);
            a.ident = parseFloat(obj['Ident']);
            a.pain = parseFloat(obj['Pain']);
            a.risk = parseFloat(obj['Risk']);
            a.target_classes = obj['TargetClasses'];
            a.time_used_occupied = parseFloat(obj['TimeUsedOccupied']);
            a.time_used_process = parseFloat(obj['TimeUsedProcess']);

            return a;
        }

    }

}